sda-download/api/sda/sda.go

```diff
@@ -29,6 +37,76 @@ func sanitizeString(str string) string {
 	return pattern.ReplaceAllString(str, "[identifier]: $1")
 }
 
+func reencryptHeader(oldHeader []byte, reencKey string) ([]byte, error) {
+	var opts []grpc.DialOption
+	if config.Config.Reencrypt.CACert != "" {
+		cacert := config.Config.Reencrypt.CACert
+		clientKey := config.Config.Reencrypt.ClientKey
+		clientCert := config.Config.Reencrypt.ClientCert
+		rootCAs := x509.NewCertPool()
+		cacertByte, err := os.ReadFile(cacert)
+		if err != nil {
+			log.Errorf("Failed to read CA certificate, reason: %s", err)
+
+			return nil, err
+		}
+		ok := rootCAs.AppendCertsFromPEM(cacertByte)
+		if !ok {
+			log.Errorf("Failed to append CA certificate to rootCAs")
+
+			return nil, errors.New("failed to append CA certificate to rootCAs")
+		}
+		// Load the client key pair
+		certs, err := tls.LoadX509KeyPair(clientCert, clientKey)
+		if err != nil {
+			log.Errorf("Failed to load client key pair for reencrypt, reason: %s", err)
+			log.Debugf("clientCert: %s, clientKey: %s", clientCert, clientKey)
+
+			return nil, err
+		}
+		clientCreds := credentials.NewTLS(
+			&tls.Config{
+				Certificates: []tls.Certificate{certs},
+				MinVersion:   tls.VersionTLS13,
+				RootCAs:      rootCAs,
+			},
+		)
+		// Use secure gRPC connection with mutual TLS authentication
```

Comment:
Tying this to the presence of a CA certificate is wrong, since the CA root might be public and thus doesn't need to be configured.
Also all these checks and reading/parsing of the certificate files should be done as part of the startup procedure *not* when a request comes in.

Reply:
Thank you @jbygdell for pointing out this problem. Considering it will be Easter soon, and to avoid the this PR stays for a long time, I would suggest we solve the TLS problem for `connect reencrypt to sda-download` in a new issue.
What do others think? @MalinAhlberg @aaperis @pahatz @pontus ?

Reply:
I'm not sure of the exact implications of this problem, but I tend to agree that it could be moved to a new issue. Or, if @jbygdell sits on a quick fix, maybe you could contribute with a commit to this PR? (also ping to @dbampalikis )